app/backend/src/tests/test_email.py

```diff
@@ -124,3 +95,113 @@ def mock_send_email(sender_name, sender_email, recipient, subject, plain, html):
 
         with patch("couchers.email.send_email", mock_send_email):
             send_report_email(complaint)
+
+
+def test_host_request_email(db):
+    with session_scope(db) as session:
+        from_user, api_token_from = generate_user(db)
+        to_user, api_token_to = generate_user(db)
+        from_date = "2020-01-01"
+        to_date = "2020-01-05"
+
+        conversation = Conversation()
+        message = Message()
+        message.conversation_id = conversation.id
+        message.author_id = from_user.id
+        message.text = random_hex(64)
+
+        host_request = HostRequest(
+            conversation_id=conversation.id,
+            from_user=from_user,
+            to_user=to_user,
+            from_date=from_date,
+            to_date=to_date,
+            status=HostRequestStatus.pending,
+            from_last_seen_message_id=message.id,
+        )
+
+        message_id = random_hex(64)
+
+        def mock_send_email(sender_name, sender_email, recipient, subject, plain, html):
+            assert recipient == to_user.email
+            assert "host request" in subject.lower()
+            assert to_user.name in plain
+            assert to_user.name in html
+            assert from_user.name in plain
+            assert from_user.name in html
+            assert from_date in plain
+            assert from_date in html
+            assert to_date in plain
+            assert to_date in html
+            assert from_user.avatar_url not in plain
+            assert from_user.avatar_url in html
+            assert f"{config['BASE_URL']}/hostrequests/" in plain
+            assert f"{config['BASE_URL']}/hostrequests/" in html
+            return message_id
+
+        with patch("couchers.email.send_email", mock_send_email):
+            send_host_request_email(host_request)
```

Comment:
verify that our mock function was called once.

Reply:
Good idea

Reply:
Cool, didn't know about create_autospec before :)